modules/packages/nimclient

```diff
@@ -0,0 +1,193 @@
+    #   that package that package should be removed from the cache
+    # - If you don't then mission portal may show that there are updates
+    #   available that are actually already installed (until the cache gets
+    #   refreshed)
+
+    # Since we don't yet know how to determine which packages have updates
+    # available we simply return true.
+    /usr/bin/true
+}
+
+list_updates_local () {
+    # This function should return the cached list of package updates availabel
+    # IF there is no cache then it should return nothing
+    # This function should avoid hitting the network for listing
+    # returns same info as list_updates
+    # CFEngine determines which one to call based on if_elapsed
+
+    # see if showres can do offline listing, see if we can know which filesets are considered updates
+    # - only list the latest update
+    /usr/bin/true
+}
+
+get_package_data () {
+    # NIM is only a REPO type install
+    # - Could add file based install for bff or rpm packages
+
+    #if echo "${File}" | grep '^/' >/dev/null; then
+        # If there's a / in $File then we'll expec this to be a 'file' install.
+        # First we need to figure out if the package matches .bff or .rpm
+        #  - If not fail
+        #    - fatal "Unsupported Package Type"
+        # Next we need to query the package for the base name and version
+        # Finally spit out the stuff
+        # echo "PackageType=file"
+        # echo NAME=
+        # echo VERSION=
+        # echo ARCHITECUTE=
+        #echo "Name=$(echo "$File" | sed 's/.*\///g')"
+    #else
+        # If $File does not contain /, it must be in an existing remote repo
+        echo "PackageType=repo"
+        echo "Name=${File}"
+}
+
+parse_pkg_data () {
+    # Emit package name and version, and arch based on output from nimclient
+    # showres. If file based install support is added then this will need to be
+    # improved to handle that case.
+    name=$(echo $1 | awk -F':' '{ print $2}')
+    version=$(echo $1 | awk -F':' '{ print $3 }')
+
+    echo "Name=$name"
+    echo "Version=$version"
+    # ARCH is useless on AIX?
+    echo "Architecture=PPC"
+}
+
+# Cfengine passes data on STDIN. Absorb that and convert to shell variables.
+while IFS= read -r line; do
+  eval "$line"
+  # options can be passed multiple times so we need to avoid clobbering
+  # previous instances. Plus, what we really want to eval is the value of
+  # each option so that we can have a variable for each value.
+  # For example options => { "lpp_source=aix7783" }
+  #   comes through the protocol as options=lpp_source=aix7783
+  #   and here we define lpp_source=aix7783
+  if [ -n "$options" ]; then
+    eval $options
+  fi
+done
+
+# Set up mock environment if necessary
+# This is not well developed as I don't have continuous access to aix and nim
+# nor am I an expert
+CFENGINE_TEST_NIMCLIENT_MOCK=false
+if [ -n "$CFENGINE_TEST_PKGSRC_MOCK" ]; then
```

Comment:
These two variables are different, but I assume they should be the same? In that case the `if [ -n` check works only by accident, you should probably correct this to:

```
CFENGINE_TEST_NIMCLIENT_MOCK=false
if [ "$CFENGINE_TEST_NIMCLIENT_MOCK" = "true" ]; then
```